niceman/resource/ssh.py

```diff
@@ -79,7 +81,20 @@ def connect(self, password=None):
                   self._connection.port,  # Fabric defaults to 22.
                   auth)
 
-        self._connection.open()
+        try:
+            self._connection.open()
+        except AuthenticationException:
+            # Give the user a chance to enter the password again.
```

Comment:
I was a bit confused by this comment, particularly "again".  AFAICT the user only has one chance to enter a password.  The only time there would already be a password is if the `password` parameter was specified, but that is only relevant for testing, right?  

So I think "again" should be dropped, and it should be made clearer that the `if password` bit is currently for testing purposes, either by tweaking the message  or adding a comment.

Reply:
Agreed. Neither the comment nor the if statement are necessary. Will remove them both.